internal/terraform/aws/ec2_instance_test.go

```diff
@@ -0,0 +1,108 @@
+package aws_test
+
+import (
+	"infracost/internal/testutil"
+	"infracost/pkg/costs"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/shopspring/decimal"
+)
+
+func TestEc2InstanceIntegration(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping test in short mode")
+	}
+
+	tf := `
+resource "aws_instance" "instance1" {
+	ami           = "fake_ami"
+	instance_type = "m3.medium"
+
+	root_block_device {
+		volume_size = 10
+	}
+
+	ebs_block_device {
+		device_name = "xvdf"
+		volume_size = 10
+	}
+
+	ebs_block_device {
+		device_name = "xvdg"
+		volume_type = "standard"
+		volume_size = 20
+	}
+
+	ebs_block_device {
+		device_name = "xvdh"
+		volume_type = "sc1"
+		volume_size = 30
+	}
+
+	ebs_block_device {
+		device_name = "xvdi"
+		volume_type = "io1"
+		volume_size = 40
+		iops        = 1000
+	}
+}`
+
+	resourceCostBreakdowns, err := testutil.RunTFCostBreakdown(tf)
+	if err != nil {
+		t.Error(err)
+	}
+
+	expectedPriceHashes := [][]string{
+		{"aws_instance.instance1", "instance hours (m3.medium)", "666e02bbe686f6950fd8a47a55e83a75-d2c98780d7b6e36641b521f1f8145c6f"},
```

Comment:
In the future, it might help to add a note to the contributing section to explain our thinking behind the integration tests and product/price hashes.